usdt-impl/src/stapsdt/args.rs

```diff
@@ -17,83 +17,35 @@
 use crate::DataType;
 use dtrace_parser::{BitWidth, DataType as NativeDataType, Integer, Sign};
 
-/// Convert an Integer type and a register index into a GNU Assembler operation
-/// that reads the integer's value from the correct register. Effectively this
-/// means generating a string like `%REG` where `REG` is the register that the
-/// data is located in.
-fn integer_to_asm_op(integer: &Integer, reg_index: u8) -> &'static str {
+/// Convert an Integer type and an argument index into a GNU Assembler
```

Comment:
why GNU assembler? Perhaps a unit test for this would make its operation clearer?

Reply:
Added unit tests for formatting arguments and the `integer_to_asm_op` fn.